### PR TITLE
Add dropdown component

### PR DESCRIPTION
**Pull request by allanpope**
**What**

* User has control over how dropdown items appear and what the elements are
* Uses same shadow(Ben wanted to keep this consistent)
* Red on hover, darker red on focus

### CODE REVIEW

File: packages/components/src/Dropdown/Dropdown.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      <Dropdown isOpen>
+        Options
+        <Dropdown.item bg="red" foo="bar">Edit</Dropdown.item>
+        <Dropdown.item>Delete</Dropdown.item>
+        <Dropdown.item>Unpublish</Dropdown.item>
+        <Dropdown.item>Duplicate</Dropdown.item>
+      </Dropdown>
+    );
+    wrapper = shallowWithTheme(component, theme).dive();
+  });
```
Comment (angusfretwell) [inline packages/components/src/Dropdown/Dropdown.test.js]: Might be worth having a snapshot of it closed as well as open.

File: packages/components/src/Dropdown/Dropdown.story.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { withDocs } from 'storybook-readme';
+import { Box, Icon, Link } from '..';
+
+import Dropdown from './Dropdown';
+import README from './README.md';
+
+storiesOf('Components|Dropdown', module)
+  .addDecorator(withDocs(README))
+  .add('default', () => (
+    <Box width="200px">
```
Comment (angusfretwell) [inline packages/components/src/Dropdown/Dropdown.story.js]: What happens without the `Box`, does it go full width or something? If so, maybe the dropdown needs to be `display: inline-block` or something?

File: packages/components/src/Dropdown/Dropdown.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  ${props => props.highlighted && css`
+    color: ${themeGet('colors.grey.0')};
+    font-weight: bold;
+  `}
+`;
+
+Dropdown.item.displayName = 'Dropdown.item';
+
+Dropdown.item.defaultProps = {
+  blacklist: ['highlighted'],
```
Comment (angusfretwell) [inline packages/components/src/Dropdown/Dropdown.js]: Do you get an error when passing `bg` or something like that?

File: packages/components/src/Dropdown/Dropdown.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+Dropdown.item = NakedButton.extend`
+  padding: ${themeGet('space.2')} ${themeGet('space.8')};
+  color: ${themeGet('colors.grey.1')};
+  width: 100%;
+  display: block;
+  text-align: left;
+
+  ${props => props.highlighted && css`
+    color: ${themeGet('colors.grey.0')};
+    font-weight: bold;
```
Comment (angusfretwell) [inline packages/components/src/Dropdown/Dropdown.js]: Should use the theme value here.

File: packages/components/src/Dropdown/Dropdown.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      isOpen,
+      getToggleButtonProps,
+      highlightedIndex,
+      getItemProps,
+      getRootProps,
+    }) => {
+    const [items, toggle] = partition(
+    React.Children.toArray(children),
+        child => child.type === Dropdown.item,
+    );
```
Comment (angusfretwell) [inline packages/components/src/Dropdown/Dropdown.js]: Indentation looks a bit strange here

File: packages/components/src/Dropdown/Dropdown.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    return (
+      <Wrapper {...getRootProps({ refKey: 'innerRef' })}>
+        <Toggle
+          {...getToggleButtonProps()}
+        >
+          {toggle}
+          <Text hidden>Toggle Dropdown</Text>
```
Comment (angusfretwell) [inline packages/components/src/Dropdown/Dropdown.js]: I wonder if we should have a prop like `accessibilityLabel` or whatever to override this? Just thinking we might want to be more explicit in some places?

Comment (allanpope) [reply packages/components/src/Dropdown/Dropdown.js]: Yeah good point, you probably would want to be explicit in most places

File: packages/components/src/Dropdown/Dropdown.js (hunk 3 above)
Comment (allanpope) [reply packages/components/src/Dropdown/Dropdown.js]: Nothing happens, `bg=red` gets added to `dropdown` but it does not throw an error.

Comment (angusfretwell) [reply packages/components/src/Dropdown/Dropdown.js]: Sweet, I meant to ask, does `bg` get added as an attribute onto the `button`?

Comment (angusfretwell) [reply packages/components/src/Dropdown/Dropdown.js]: And like the same for `mt`, etc.

Comment (allanpope) [reply packages/components/src/Dropdown/Dropdown.js]: If you add props they only get added to `<Downshift />`, good point though.

Right now you could wrap a `Box` around it with `mt` to move its position. 
You could also add a `Box` as a child to change the background color. You wouldnt be able to change the hover/focus states though.

Do you think its needed?

**Pull request merged**